src/components/SelectButtonGroup/SelectButtonGroup/SelectButtonGroup.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import bem from '../../../utils/bem';
+import { SelectButtonProps } from '../SelectButton';
+import styles from './SelectButtonGroup.scss';
+
+interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+    /** SelectButton components */
+    children: React.ReactElement<SelectButtonProps>[];
+    /**
+     * A function to be called selection is changed.
+     * For convenience it will always be called with an array. In case of single select this array will always be of max length 1
+     */
+    onChange?: (selection: string[]) => void;
+    /** is this button part of a multiselect group - when yes will allow more then one option to be selected */
+    isMultiselect?: boolean;
+    /** should the component take up all the width available */
+    isBlock?: boolean;
+    /** Color context for selected buttons */
+    selectedContext?: 'neutral' | 'brand';
+    /** should children have equal width */
+    isEqualWidth?: boolean;
+}
+
+const { block } = bem('SelectButtonGroup', styles);
+
+const SelectButtonGroup: React.FC<Props> = props => {
+    const {
+        children,
+        isMultiselect,
+        isEqualWidth,
+        isBlock,
+        selectedContext,
+        onChange,
+        ...rest
+    } = props;
+
+    const initiallySelectedValues: string[] = [];
+    children.forEach(child => {
+        const { value, isSelected } = child.props;
+        if (isSelected) {
+            initiallySelectedValues.push(value);
```

Comment:
can be replaced with `initiallySelectedValues = [... initiallySelectedValues, value];` to avoid using of mutating `push` method

Reply:
Why is it wrong to use push here, when this array was created only a few lines above only for the reason of holding these values?